sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -99,6 +106,71 @@
         }
     }
 
+    /**
+     * Asynchronously acquire a token from Active Directory with Aure CLI.
+     *
+     * @param request the details of the token request
+     * @return a Publisher that emits an AccessToken
+     */
+    public Mono<AccessToken> authenticateWithAzureCli(TokenRequestContext request) {
+        String azCommand = "az account get-access-token --output json --resource ";
+    
+        StringBuilder command = new StringBuilder();
+        command.append(azCommand);
+        String scopes = ScopeUtil.scopesToResource(request.getScopes());
+        command.append(scopes);
+    
+        AccessToken token = null;
+        try {
+            String starter;
+            String switcher;
+            if (System.getProperty("os.name").contains("Windows")) {
+                starter = "cmd.exe";
```

Comment:
Use string constants.

Reply:
https://github.com/Azure/azure-sdk-for-java/blob/41a932e2bc316f58334cd943499991572c283266/sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java#L495
In the method `openUrl` of class `IdentityClient`, doing similar things with string, didn't use constants. Is it better to follow that way?

Reply:
I would suggest creating `private static final String WINDOWS_STARTER = "cmd.exe"` and then use `starter = WINDOWS_STARTER` here.